File: programmers/java/0022. 두 정수 사이의 합.java
class Solution {
    public long solution(int a, int b) {
        int tmp;
        if (a > b) {tmp = a; a = b; b = tmp;}
        return sum(a,b);
    }
    private long sum(long a, long b){
        return (b - a + 1)*(a + b)/2;
    }
}

File: programmers/cpp/0022. 두 정수 사이의 합.cpp
#include <algorithm>

using namespace std;

long long solution(int a, int b) {
    long long answer = 0;
    int na, nb;
    na = min(a,b);
    nb = max(a,b);
    return (long long)(nb - na + 1)*(na + nb)/2;
}
